### PR TITLE
Add image/video inserting to RCE

### PR DESCRIPTION
**Pull request by vanwagonet**
Files are immediately uploaded, with progress shown inline.

refs: MBL-11299
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  One or more files in Core are below the minimum test coverage 50%
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-11299](https://instructure.atlassian.net/browse/MBL-11299)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 92.05% | 93.62% | -1.57%
Core/Core/RichContentEditor/RichContentEditorPresenter.swift | 0% | -- | --
**Student** | 92.17% | 92.17% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.63% | 92.63% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/RichContentEditor/RichContentEditorPresenter.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    func insertImagePlaceholder(_ url: URL, placeholder: String)
+    func insertVideoPlaceholder(_ url: URL)
+    func updateUploadProgress(of files: [File])
+}
+
+public class RichContentEditorPresenter: NSObject, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    let batchID = UUID.string
+    var env: AppEnvironment = .shared
+    let uploadContext: FileUploadContext
+    var uploader: FileUploader = UploadFile(bundleID: Bundle.main.bundleIdentifier ?? Bundle.studentBundleID, appGroup: Bundle.main.appGroupID())
```
[the remark this answers is not in the document]
Comment (vanwagonet) [reply Core/Core/RichContentEditor/RichContentEditorPresenter.swift]: I suppose so, but the main bundle should never be missing a `bundleIdentifier`.

File: Core/Core/RichContentEditor/RichContentEditorViewController.swift [hunk 2]
```diff
@@ -14,19 +14,33 @@
...
-import WebKit
+import MobileCoreServices
 import UIKit
+import WebKit
 
 public protocol RichContentEditorDelegate: class {
-    func rce(_ editor: RichContentEditorViewController, didChangeEmpty isEmpty: Bool)
+    func rce(_ editor: RichContentEditorViewController, canSubmit: Bool)
+    func rce(_ editor: RichContentEditorViewController, didError error: Error)
+    func rce(_ editor: RichContentEditorViewController, didError error: String)
```
[the remark this answers is not in the document]
Comment (vanwagonet) [reply Core/Core/RichContentEditor/RichContentEditorViewController.swift]: Yeah, that makes more sense.

**Comment by vanwagonet** [comment]
I still need to write tests for the presenter, but thought I'd get a build into peoples hands sooner rather than later.

**Pull request merged**